django/db/models/sql/compiler.py

```diff
@@ -979,6 +980,12 @@ def get_default_columns(
         # be used by local fields.
         seen_models = {None: start_alias}
 
+        select_mask_fields = set()
+        for field in select_mask:
+            select_mask_fields.update(
+                field.fields if isinstance(field, CompositePrimaryKey) else [field]
```

Comment:
I guess that'll eventually be some form of check for `field.composite` once/when we support something else than only `CompositePrimaryKey`.

Reply:
Yeah I'm not a fan of `isinstance` checks either, I suppose we could check if `field` has a `fields` attribute to make this a little more versatile.

Reply:
This has been updated to support the above use-case - I'm resolving the thread now.